File: java/leetcode/src/Task146.java
import org.junit.jupiter.api.Test;

import java.util.*;
class Task146{
    class LRUCache {

        LinkedHashMap<Integer,Integer> map;
        int capacity;

        public LRUCache(int capacity) {
            map = new LinkedHashMap<>(capacity, 0.75f,true);
            this.capacity = capacity;
        }

        public int get(int key) {
            return map.getOrDefault(key,-1);
        }

        public void put(int key, int value) {
            map.put(key,value);

            if(map.size() > capacity){
                int removeKey = map.keySet().iterator().next();
                map.remove(removeKey);
            }

        }
        /**
         * Your LRUCache object will be instantiated and called as such:
         * LRUCache obj = new LRUCache(capacity);
         * int param_1 = obj.get(key);
         * obj.put(key,value);
         */



    }

    @Test
    void test(){
        LRUCache cache = new LRUCache(2);
        cache.put(2, 1);
        cache.put(1, 2);
        cache.put(2, 3);
        cache.put(4, 1);
        System.out.println(cache.get(1));
        System.out.println(cache.get(2));

    }
}

File: py/Task146.py

class LRUCache:

    def __init__(self, capacity: int):
        self.capacity = capacity
        self.map = {}

    def get(self, key: int) -> int:
        if key in self.map:
            val = self.map[key]
            del self.map[key]
            self.map[key] = val
            return val
        else:
            return -1

    def put(self, key: int, value: int) -> None:
        if key in self.map:
            del self.map[key]
        self.map[key] = value
        if len(self.map) > self.capacity:
            it = iter(self.map)
            del self.map[next(it)]

cache = LRUCache(2)

cache.put(1, 1)
cache.put(2, 2)
cache.get(1)
cache.put(3, 3)
cache.get(2)
cache.put(4, 4)
cache.get(1)
cache.get(3)
cache.get(4)
